### ISSUE TITLE
Line length is mesured in bytes instead of characters.

### ISSUE THREAD

**Issue by AJenbo** [squizlabs/PHP_CodeSniffer#1300]
The PSR2 rule responsible for checking line length incorrectly reports excessively long lines when the line contains multi-byte characters.

test.php:
`<?php // This line is 61 characters long 💩💩💩💩💩💩💩💩💩💩💩💩💩💩💩💩💩💩💩💩`

`phpcs --standard=PSR2 "test.php"`

Will result in the following warning:

> test.php | 1 | WARNING | Line exceeds 120 characters; contains 121 characters

**Comment by jrfnl** [comment]
As far as I can see, this is not so much related to the sniff, but to the token position map.
That function uses the `iconv_strlen()` function to determine the length of a token if available and falls back to `strlen()` if not.
I presume the PHP version you are using does not have the iconv extension available ?

**Comment by aik099** [comment]
Or you're not specifying `--encoding=UTF-8` when running `phpcs` or `phpcbf`.

**Comment by gsherwood** [comment]
The encoding option is probably your problem. From version 3.0, UTF8 encoding is the default, but you need to specify it on the command line, or in a ruleset XML file, in older versions.

Did setting the encoding work for you?

**Comment by Ovsyanka** [comment]
I faced with same situation and yes, `--encoding=UTF-8` solved this. Thank you!

**Comment by gsherwood** [comment]
Closing due to no feedback. I assume the suggested fix will work given it was verified by a few people.

**Comment by mplx** [comment]
PHP 7.2.9, PHP_CodeSniffer 3.3.1

code line w/ length 114:
```
                    ->with('XXXXXX XXXX äöü', 'XXXXXXX X äöü', 'XXXXXXX X äöü', 'XXXXXXX X äöü', $xxxXxx)
```

`phpcs --standard=PSR2 sample.php -p`

php with iconv module enabled:  
no errors

php without iconv module:  
`WARNING | Line exceeds 120 characters; contains 125 characters`

`phpcs --encoding=UTF-8 --standard=PSR2 sample.php -p`

php with iconv module enabled:  
no errors

php without iconv module:  
`WARNING | Line exceeds 120 characters; contains 125 characters`
